uxarray/grid/integrate.py

```diff
@@ -0,0 +1,468 @@
+
+    latitude_cart : float
+        The latitude in Cartesian coordinates (The normalized z coordinate)
+
+    face_latlon_bound : np.ndarray
+        The list of latitude and longitude bounds of faces. Shape: (n_faces,2, 2),
+        [...,[lat_min, lat_max], [lon_min, lon_max],...]
+
+    is_directed : bool, optional (default=False)
+        If True, the GCA is considered to be directed, which means it can only from v0-->v1. If False, the GCA is undirected,
+        and we will always assume the small circle (The one less than 180 degree) side is the GCA.
+
+    is_latlonface : bool, optional, default=False
+        A global flag to indicate if faces are latlon face. If True, then treat all faces as latlon faces. Latlon face means
+        That all edge is either a longitude or constant latitude line. If False, then all edges are GCA.
+         Default is False.
+
+    is_face_GCA_list : np.ndarray, optional (default=None)
+        A list of boolean values that indicates if the edge in that face is a GCA. Shape: (n_faces,n_edges).
+        True means edge face is a GCA.
+        False mean this edge is a constant latitude.
+        If None, all edges are considered as GCA. This attribute will overwrite the is_latlonface attribute.
+
+    Returns
+    -------
+    weights_df : pandas.DataFrame
+        A DataFrame with the calculated weights of each face. The DataFrame has two columns:
+        - 'face_index': The index of the face (integer).
+        - 'weight': The calculated weight of the face in radian (float).
+        The DataFrame is indexed by the face indices, providing a mapping from each face to its corresponding weight.
+    """
+    intervals_list = []
+
+    # Iterate through all faces and their edges
+    for face_index, face_edges in enumerate(faces_edges_cart):
+        if is_face_GCA_list is not None:
+            is_GCA_list = is_face_GCA_list[face_index]
+        else:
+            is_GCA_list = None
+        face_interval_df = _get_zonal_face_interval(
+            face_edges,
+            latitude_cart,
+            face_latlon_bound[face_index],
+            is_directed=is_directed,
+            is_latlonface=is_latlonface,
+            is_GCA_list=is_GCA_list,
+        )
+        for _, row in face_interval_df.iterrows():
+            intervals_list.append(
+                {"start": row["start"], "end": row["end"], "face_index": face_index}
+            )
+
+    intervals_df = pd.DataFrame(intervals_list)
+    overlap_contributions, total_length = _process_overlapped_intervals(intervals_df)
+
+    # Calculate weights for each face
+    weights = {
+        face_index: overlap_contributions.get(face_index, 0.0) / total_length
+        for face_index in range(len(faces_edges_cart))
+    }
+
+    # Convert weights to DataFrame
+    weights_df = pd.DataFrame(list(weights.items()), columns=["face_index", "weight"])
+    return weights_df
+
+
+def _get_zonal_face_interval(
```

Comment:
This function is extremely long, with multiple if-else blocks that could be turned into helper functions. Can you turn some of those if-else blocks into helper functions?